### ISSUE TITLE
Process.set_priority and Process.get_priority

### ISSUE THREAD

**Issue by TheMasterCado** [crystal-lang/crystal#8828]
After a post on the forum, I've been advised to post an issue here to discuss about adding this in the standard library. 

Here's a link to the forum post: [https://forum.crystal-lang.org/t/getting-and-setting-process-priority-nice-value/1736](https://forum.crystal-lang.org/t/getting-and-setting-process-priority-nice-value/1736)

This is about adding a `get_priority` and `set_priority` function to set and get the niceness of a process just like in Ruby.

**Comment by straight-shoota** [comment]
Process niceness is a pretty platform-specific feature. I assume a similar concept exists everywhere, but I'm not sure how feasible it is to create a portable API.

**Comment by yxhuvud** [comment]
How does other languages handle the question? What do other languages that try to keep platform independent APIs (like Java. any other that are low enough level to care?) do?

**Comment by RX14** [comment]
I'd really rather leave the most pletform-specific `Process` stuff in a blessed `crystal-lang/unix` shard or similar.

**Comment by TheMasterCado** [comment]
As I know all Linux systems can handle this, no?

**Comment by TheMasterCado** [comment]
@RX14 

I'm not very familiar with the Crystal project (I use the language and that's about it), but what other platform would be a problem if LibC is used ?

**Comment by Sija** [comment]
Windows has priority that's pretty similar, see http://code.activestate.com/recipes/496767-set-process-priority-in-windows/

**Comment by TheMasterCado** [comment]
@Sija Yes I see, but according to the Wiki, the Windows platform is not officially supported, but I understand it may be planned.

So we could add something similar for windows. The way we set priority could be dependent on the platform?

**Comment by straight-shoota** [comment]
Windows support is in progress. Every API in stdlib needs to be portable.

**Comment by TheMasterCado** [comment]
Yes and it's better like that, but as @Sija said, there is a similar priority for processes in Windows, wouldn't it be possible to get the getpriority and setpriority methods to act differently depending on the platform?
